common/maplib/map.mjs

```diff
@@ -438,6 +470,29 @@ class _SingleGameMap {
     if (tile === null) return null;
     return this.graphicAsset.getTile(layer, tile);
   }
+
+  /**
+   * Get spawn points of single map.
+   * If We have a spawn point area from (1,1) (1,5) (5,1) (5,5),
+   * we split it here to become 16 spawn points.
+   * @return {Array} The map coordinates inside 'spawnPoint' cell set.
+   */
+  getSpawnPoints() {
+    if (typeof this._spawnPointList === 'undefined') {
+      if (this.staticCellSet.has('spawnPoint')) {
+        for (const cell in this.staticCellSet.get('spawnPoint')) {
+          const w = (cell.w ?? 1);
+          const h = (cell.h ?? 1);
+          for (let i = 0; i < w; ++i) {
+            for (let j = 0; j < h; ++j) {
+              this._spawnPointList.push(new MapCoord(this.gameMap, cell.x + i, cell.y + j));
+            }
+          }
+        }
+      }
```

Comment:
Is this error expected? I didn't see where the `staticCellSet` variable was declared.

```
Player yKSkP connected.
file:///..../hitcon-online/common/maplib/map.mjs:483
      if (this.staticCellSet.has('spawnPoint')) {
                             ^

TypeError: Cannot read property 'has' of undefined
    at _SingleGameMap.getSpawnPoints (file:///..../hitcon-online/common/maplib/map.mjs:483:30)
    at file:///..../hitcon-online/common/maplib/map.mjs:258:42
    at Map.forEach (<anonymous>)
    at GameMap.getRandomSpawnPointNoStarvation (file:///..../hitcon-online/common/maplib/map.mjs:257:18)
    at GatewayService.addSocket (file:///..../hitcon-online/services/gateway/gateway-service.mjs:194:47)
```

Reply:
Yes, I forgot I rebase last time. Wait me a minute !